Assets/Scripts/StandardSamples/UI/Achievements/UIAchievementsMenu.cs

```diff
@@ -239,6 +162,28 @@ private async void OnAchievementDataUpdated()
             RefreshDisplayingDefinition();
         }
 
+        private async Task AddAchievementButton(AchievementData achievement)
+        {
+            string achievementId = achievement.Definition.AchievementId;
+            
+            var button = Instantiate(itemTemplate, achievementListContainer);
+            button.SetNameText(achievementId);
+            button.gameObject.SetActive(true);
+            button.index = achievementListItems.Count;
+
+            achievementListItems.Add(button);
+
+            bool unlocked = achievement.PlayerData.HasValue && achievement.PlayerData.Value.Progress >= 1;
+
+            Task<Texture2D> getIconTextureTask = unlocked
+                ? AchievementsService.Instance.GetAchievementUnlockedIconTexture(achievementId)
+                : AchievementsService.Instance.GetAchievementLockedIconTexture(achievementId);
+
+            var tex = await getIconTextureTask;
```

Comment:
This makes me think we should have a temporary "loading" icon. Perhaps whatever is already set in the prefab, which currently will look like a white square.

Reply:
The way our current implementation works is that it has an empty white square as you mentioned. A "loading" icon couldn't hurt, but is outside the scope of the PR.

Reply:
Added #EOS-2024 for this.